rpc/types/src/misc/tx_entry.rs

```diff
@@ -18,42 +20,127 @@ use cuprate_epee_encoding::{
-    pub received_timestamp: u64,
-    pub relayed: bool,
-    pub tx_hash: String,
+#[cfg_attr(feature = "serde", serde(untagged))]
+pub enum TxEntry {
+    /// This entry exists in the transaction pool.
+    InPool {
+        as_hex: String,
+        as_json: String,
+        block_height: u64,
+        block_timestamp: u64,
+        confirmations: u64,
+        double_spend_seen: bool,
+        output_indices: Vec<u64>,
+        prunable_as_hex: String,
+        prunable_hash: String,
+        pruned_as_hex: String,
+        tx_hash: String,
+        #[cfg_attr(feature = "serde", serde(serialize_with = "serde_true"))]
+        /// Will always be serialized as `true`.
+        in_pool: bool,
+    },
+    /// This entry _does not_ exist in the transaction pool.
+    NotInPool {
+        as_hex: String,
+        as_json: String,
+        double_spend_seen: bool,
+        prunable_as_hex: String,
+        prunable_hash: String,
+        pruned_as_hex: String,
+        received_timestamp: u64,
+        relayed: bool,
+        tx_hash: String,
+        #[cfg_attr(feature = "serde", serde(serialize_with = "serde_false"))]
+        /// Will always be serialized as `false`.
+        in_pool: bool,
+    },
+}
+
+impl Default for TxEntry {
+    fn default() -> Self {
+        Self::NotInPool {
+            as_hex: String::default(),
+            as_json: String::default(),
+            double_spend_seen: bool::default(),
+            prunable_as_hex: String::default(),
+            prunable_hash: String::default(),
+            pruned_as_hex: String::default(),
+            received_timestamp: u64::default(),
+            relayed: bool::default(),
+            tx_hash: String::default(),
+            in_pool: false,
+        }
+    }
 }
 
-// TODO: custom epee
-// <https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server_commands_defs.h#L406-L427>
+//---------------------------------------------------------------------------------------------------- Epee
 #[cfg(feature = "epee")]
-epee_object! {
-    TxEntry,
-    as_hex: String,
-    as_json: String, // TODO: should be its own struct
-    block_height: u64,
-    block_timestamp: u64,
-    confirmations: u64,
-    double_spend_seen: bool,
-    in_pool: bool,
-    output_indices: Vec<u64>,
-    prunable_as_hex: String,
-    prunable_hash: String,
-    pruned_as_hex: String,
-    received_timestamp: u64,
-    relayed: bool,
-    tx_hash: String,
+impl EpeeObjectBuilder<TxEntry> for () {
+    fn add_field<B: Buf>(&mut self, name: &str, r: &mut B) -> error::Result<bool> {
+        unreachable!()
+    }
+
+    fn finish(self) -> error::Result<TxEntry> {
+        unreachable!()
+    }
+}
+
+#[cfg(feature = "epee")]
+impl EpeeObject for TxEntry {
+    type Builder = ();
+
+    fn number_of_fields(&self) -> u64 {
+        unreachable!()
+    }
+
+    fn write_fields<B: BufMut>(self, w: &mut B) -> error::Result<()> {
+        unreachable!()
+    }
 }
```

Comment:
Invariant documented in `lib.rs` - some types will be unreachable and/or not impl for wrong context, e.g. `TxEntry` here is only used in a JSON endpoint, so its epee impl just panics.

This is a workaround because the macro requires all types impl serde + epee.